### PR TITLE
Prometheus and Grafana supports added.

### PR DESCRIPTION
**Pull request by sonerpyci**
Prometheus and Grafana configuration files and docker-compose services were implemented. 
One of the required endpoints for monitoring(/metrics) and one of the dummy endpoints for monitoring tests(/slow) were added to app.ts.

The Image below contains some example metrics collected on my local system.
To show similar graphs, users should create them at grafana which will be available on http://localhost:8080/ with required queries.

Here is my queries to create charts :

1.         Request Per Minutes : sum(rate(http_request_duration_seconds_count[1m])) by (route, method, code)  * 60
2.         Request Per Seconds : sum(rate(http_request_duration_seconds_count[2m])) by (route)
3.         Memory Usage In Mb's : avg(nodejs_external_memory_bytes / 1024)
4.         CPU Usage

             - Cpu Usage : sum(avg by (instance,mode) (irate(node_cpu_seconds_total{mode!='idle'}[1m]))) * 100
             - Cpu Free : sum(avg by (instance,mode) (irate(node_cpu_seconds_total{mode='idle'}[1m]))) * 100 

image

As I mentioned in the related [issue](https://github.com/evrimagaci/periodum-api/issues/5) with using different metrics and queries, different charts can be created to monitor some other resources as well.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -16,9 +24,68 @@ services:
...
+    volumes:
+      - grafana_data:/var/lib/grafana
+      - ./grafana/provisioning:/etc/grafana/provisioning
+    environment:
+      - GF_AUTH_DISABLE_LOGIN_FORM=true
+      - GF_AUTH_ANONYMOUS_ENABLED=true
+      - GF_AUTH_ANONYMOUS_ORG_ROLE=Admin
+      - GF_SERVER_HTTP_PORT=8080
+    ports:
+      - "8080:8080"
```
Comment (nesimtunc) [inline docker-compose.yml]: Let's expose those ports only `locally`. "127.0.0.1:8080:8080"

File: docker-compose.yml [hunk 2]
```diff
@@ -16,9 +24,68 @@ services:
     depends_on:
       - db
     restart: unless-stopped
+    networks:
+      - monitoring
```
Comment (nesimtunc) [inline docker-compose.yml]: I think the network name should be `periodum`

File: docker-compose.yml [hunk 3]
```diff
@@ -16,9 +24,68 @@ services:
...
+      - /proc:/host/proc:ro
+      - /sys:/host/sys:ro
+      - /:/rootfs:ro
+    command:
+      - '--path.procfs=/host/proc'
+      - '--path.rootfs=/rootfs'
+      - '--path.sysfs=/host/sys'
+      - '--collector.filesystem.mount-points-exclude=^/(sys|proc|dev|host|etc)($$|/)'
+    ports:
+      - "9100:9100"
```
Comment (nesimtunc) [inline docker-compose.yml]: Let's expose those ports only `locally`. "127.0.0.1:9100:9100"

File: src/Monitoring/monitoring.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
+import express from "express";
+import client from "prom-client";
+
+class Monitoring {
```
Comment (nesimtunc) [inline src/Monitoring/monitoring.ts]: Let's change the folder name to lowercase please, so in that case, you may rename the file name as well to `monitoring/index.ts`

File: src/app.ts [hunk 5]
```diff
@@ -1,13 +1,28 @@
 import express, { Application, Response } from "express";
+import {monitorService} from "./Monitoring/monitoring.js";
```
Comment (nesimtunc) [inline src/app.ts]: `js`?

File: src/app.ts [hunk 6]
```diff
@@ -1,13 +1,28 @@
...
 	res.status(200).send("Hello Periodum API!");
 });
 
+
+app.get('/metrics', async (_, res) => {
+	res.setHeader('Content-Type', monitorService.register.contentType);
+	res.send(await monitorService.register.metrics());
+});
+
+app.get('/slow', async (_, res) => {
```
Comment (nesimtunc) [inline src/app.ts:19]: That's a nice case to test ;) Thanks!

**Review by nesimtunc — CHANGES_REQUESTED** [review]
Hi @sonerpyci,

Thank you for the awesome work! There're a few changes you may need to do.

File: docker-compose.yml (hunk 1 above)
Comment (sonerpyci) [reply docker-compose.yml]: Done with the last commit.

File: docker-compose.yml (hunk 2 above)
Comment (sonerpyci) [reply docker-compose.yml]: It's a good idea, "monitoring" was a forgotten one to change.
Done with the last commit.

File: docker-compose.yml (hunk 3 above)
Comment (sonerpyci) [reply docker-compose.yml]: Done with the last commit.

File: src/Monitoring/monitoring.ts (hunk 4 above)
Comment (sonerpyci) [reply src/Monitoring/monitoring.ts]: Done with the last commit as "monitoring/index.ts".

File: src/app.ts (hunk 5 above)
Comment (sonerpyci) [reply src/app.ts]: added "--experimental-specifier-resolution=node" flag to the package.json to use it without .js extension. 
Please refer https://stackoverflow.com/a/65588027

File: src/app.ts (hunk 6 above)
Comment (sonerpyci) [reply src/app.ts:19]: Thanks! :)

**Comment by nesimtunc** [comment]
Hi @sonerpyci 

Thank you for the changes! Please accept some delay from me in order to test it.

**Comment by nesimtunc** [comment]
Sorry @sonerpyci 

Due to another PR's merging this one has conflicts now, can you resolve them, please? Maybe both PRs. I guess today I'll be able to review the PRs. 

Thank you.

**Comment by nesimtunc** [comment]
Hi @sonerpyci, 

We just got AppSignal account, we'll be using it for monitoring, so, for now I'm closing this PR. The logging PR may remain and we might use it.

**Pull request closed without merging**